ocaml/lambda/translcore.ml

```diff
@@ -1369,23 +1504,16 @@ and transl_function ~in_new_scope ~scopes e alloc_mode param arg_mode arg_sort r
     end
     else enter_anonymous_function ~scopes ~assume_zero_alloc
   in
-  let arg_layout =
-    function_arg_layout e.exp_env e.exp_loc arg_sort e.exp_type
-  in
   let ((kind, params, return, region), body) =
     event_function ~scopes e
       (function repr ->
-         let pl =
-           push_defaults e.exp_loc arg_mode arg_sort cases partial warnings
+         let { params; body; return_sort; region } =
+           fuse_method_arity { params; body; return_sort; region }
```

Comment:
I think lines 1510-1511 are the same as https://github.com/ocaml/ocaml/pull/12236/files#diff-cf9720902b39bcea1aaa3c6f4ec63e8243975cb68f15410bb03d2073b59ae4c3R851. `fuse_method_arity` is certainly a bit more complicated in this PR because of modes, but I think the idea is the same.